lib/src/model/geometry_decoding.dart

```diff
@@ -24,7 +24,7 @@ _Command _decodeCommand(int command) {
 int _decodeCommandLength(int command) => command >> 3;
 
 @pragma('vm:prefer-inline')
-int _decodeZigZag(int value) => (value >> 1) ^ -(value & 1);
+int _decodeZigZag(int value) => ((value >> 1) ^ -(value & 1)).toSigned(32);
```

Comment:
Thanks for the PR, this looks fantastic.

The [vector tile spec 2.1](https://github.com/mapbox/vector-tile-spec/tree/master/2.1/) indicates: 

> A geometry is encoded as a sequence of 32 bit unsigned integers

Any concern with using signed 32-bit integers? Wondering if it might affect the results.

Also curious as to why `toSigned(32)` is applied after the bitwise operations, rather than before. Should we be doing something like this instead?

```dart
int _decodeZigZag(int value) {
  final value32 = value.toSigned(32)
  return (value32 >> 1) ^ -(value32 & 1);
}
```

Reply:
The decoded parameter values fit into signed 32-bit integers:

> Parameter values greater than pow(2,31) - 1 or less than -1 * (pow(2,31) - 1) are not supported.

The results of bitwise operations are different between native and web because of the different underlying representations. I'm not sure how this works exactly, but for 32-bit chunks we can get consistent results everywhere by interpreting the results of bit-wise operations as signed 32-bit numbers.

> For bit manipulation, consider explicitly operating on 32-bit chunks, which are consistent on all platforms. To force a signed interpretation of a 32-bit chunk, use int.toSigned(32).